src/offset_of.rs

```diff
@@ -94,6 +95,29 @@ macro_rules! offset_of {
     }};
 }
 
+#[cfg(feature = "memoffset_constant")]
+#[macro_export(local_inner_macros)]
+macro_rules! offset_of {
+    ($parent:path, $field:tt) => {{
+        const OFFSET: usize = {
+            _memoffset__field_check!($parent, $field);
+
+            // Get a base pointer.
+            // No UB here, and the pointer does not dangle, either.
+            let uninit = $crate::mem::MaybeUninit::<$parent>::uninit();
+            unsafe {
+                let base_ref = $crate::mem::transmute::<_, &$parent>(&uninit);
```

Comment:
There is UB here, this is creating a reference to uninitialized data. We just can't entirely avoid that, so a comment should say so.

On the plus side, this is inside a `const`, so we at least don't have to worry about LLVM doing anything weird.